server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -1788,7 +1788,7 @@ public void onFailure(Exception e) {
                         ThreadPool.Names.WRITE,
                         ""
                     ))
-                .isExactlyInstanceOf(AssertionError.class)
+                .hasCauseExactlyInstanceOf(AssertionError.class)
```

Comment:
This is partial revert of https://github.com/crate/crate/pull/18115 as I removed the unwrap again.